kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -253,4 +290,22 @@ private static Schema schema(Schema.Type type, boolean makeNullable) {
     }
     return Schema.Type.STRING;
   }
+  
+  private static Schema.Type stringToFieldType(String typeString) {
+    if (typeString == null || typeString.isEmpty()) {
+      return null;
+    }
+      
+    String type = typeString.toLowerCase(Locale.ENGLISH);
+    if (type.equals("long")) {
```

Comment:
This should also include int, boolean, and any other types that are supported in the CSV conversions.